modules/server/src/completion.ts

```diff
@@ -0,0 +1,339 @@
+  }
+
+  /**
+   * Get keywords relevant for the current context
+   */
+  private getKeywordsForContext(context: CompletionContext): GosuKeywordInfo[] {
+    const keywords: GosuKeywordInfo[] = []
+    
+    if (context.isFileStart) {
+      // At file start, suggest package and declaration keywords
+      keywords.push(...getKeywordsByCategory('declaration').filter(k => 
+        ['package', 'class', 'interface', 'enhancement', 'enum', 'structure'].includes(k.keyword)
+      ))
+      keywords.push(...getKeywordsByCategory('visibility'))
+    } else if (context.isInFunctionBody) {
+      // Inside function body, suggest control flow and literals
+      keywords.push(...getKeywordsByCategory('control'))
+      keywords.push(...getKeywordsByCategory('literal'))
+      keywords.push(...getKeywordsByCategory('type'))
+      keywords.push(...getKeywordsByCategory('declaration').filter(k => k.keyword === 'var'))
+    } else if (context.isInClassBody) {
+      // Inside class body, suggest member declarations
+      keywords.push(...getKeywordsByCategory('declaration').filter(k => 
+        ['function', 'var', 'property', 'construct'].includes(k.keyword)
+      ))
+      keywords.push(...getKeywordsByCategory('visibility'))
+      keywords.push(...getKeywordsByCategory('modifier'))
+    } else {
+      // Default context - return most common keywords
+      keywords.push(...getKeywordsByCategory('declaration'))
+      keywords.push(...getKeywordsByCategory('visibility'))
+    }
+    
+    return keywords
+  }
+
+  /**
+   * Convert a Gosu keyword to an LSP CompletionItem
+   */
+  private createCompletionItem(keyword: GosuKeywordInfo): CompletionItem {
+    const item: CompletionItem = {
+      label: keyword.keyword,
+      kind: this.getCompletionItemKind(keyword.category),
+      detail: keyword.description,
+      documentation: `Gosu ${keyword.category}: ${keyword.description}`,
+      insertText: keyword.snippet || keyword.keyword,
+      insertTextFormat: keyword.snippet ? InsertTextFormat.Snippet : InsertTextFormat.PlainText,
+      sortText: this.getSortText(keyword)
+    }
+    
+    return item
+  }
+
+  /**
+   * Get the appropriate LSP CompletionItemKind for a keyword category
+   */
+  private getCompletionItemKind(category: GosuKeywordCategory): CompletionItemKind {
+    switch (category) {
+      case 'declaration':
+        return CompletionItemKind.Keyword
+      case 'visibility':
+      case 'modifier':
+        return CompletionItemKind.Keyword
+      case 'control':
+        return CompletionItemKind.Keyword
+      case 'literal':
+        return CompletionItemKind.Keyword
+      case 'type':
+        return CompletionItemKind.TypeParameter
+      default:
+        return CompletionItemKind.Keyword
+    }
+  }
+
+  /**
+   * Get sort text to control completion order
+   */
+  private getSortText(keyword: GosuKeywordInfo): string {
+    // Prioritize certain keywords
+    const priority: Record<string, string> = {
```

Comment:
A floating number is the best for defining the priority, allowing one to add a digit after the comma to adjust the priority of any keyword.